Last login: Fri Mar 15 23:17:08 on ttys024
(base) beimingyouyu@beimingyouyudeMacBook-Air ~ % ssh secfuzz@shuaicpu4.cse.ust.hk
secfuzz@shuaicpu4.cse.ust.hk's password: 
Permission denied, please try again.
secfuzz@shuaicpu4.cse.ust.hk's password: 
Welcome to Ubuntu 22.04.3 LTS (GNU/Linux 6.5.0-1015-azure x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

  System information as of Sat Mar 16 04:25:24 AM UTC 2024

  System load:  2.2138671875      Processes:                740
  Usage of /:   14.3% of 1.82TB   Users logged in:          2
  Memory usage: 18%               IPv4 address for docker0: 172.17.0.1
  Swap usage:   0%                IPv4 address for enp67s0: 143.89.191.108

  => /export/d1 is using 90.7% of 7.22TB
  => /export/d2 is using 85.3% of 7.22TB
  => /export/ssd1 is using 86.9% of 1.83TB

 * Strictly confined Kubernetes makes edge and IoT secure. Learn how MicroK8s
   just raised the bar for easy, resilient and secure K8s cluster deployment.

   https://ubuntu.com/engage/secure-kubernetes-at-the-edge

Expanded Security Maintenance for Applications is not enabled.

55 updates can be applied immediately.
To see these additional updates run: apt list --upgradable

9 additional security updates can be applied with ESM Apps.
Learn more about enabling ESM Apps service at https://ubuntu.com/esm


*** System restart required ***
Last login: Fri Mar 15 14:19:07 2024 from 143.89.176.82
secfuzz@shuaicpu4:~$ docker exec -it 5d1140ea8826 /bin/bash

________                               _______________
___  __/__________________________________  ____/__  /________      __
__  /  _  _ \_  __ \_  ___/  __ \_  ___/_  /_   __  /_  __ \_ | /| / /
_  /   /  __/  / / /(__  )/ /_/ /  /   _  __/   _  / / /_/ /_ |/ |/ /
/_/    \___//_/ /_//____/ \____//_/    /_/      /_/  \____/____/|__/


WARNING: You are running this container as root, which can cause new files in
mounted volumes to be created as the root user on your host machine.

To avoid this, run the container by specifying your user's userid:

$ docker run -u $(id -u):$(id -g) args...

root@5d1140ea8826:/# cd /home/
root@5d1140ea8826:/home# source "./venv/tensorflow-2.6-orig/bin/activate"
bash: ./venv/tensorflow-2.6-orig/bin/activate: No such file or directory
root@5d1140ea8826:/home# cd ivysyn/
root@5d1140ea8826:/home/ivysyn# source "./venv/tensorflow-2.6-orig/bin/activate"
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn# python
Python 3.9.12 (main, Apr  5 2022, 06:56:58) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
2024-03-16 04:28:14.233370: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:28:14.233400: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.

>>>     tf.__version__
  File "<stdin>", line 1
    tf.__version__
IndentationError: unexpected indent
>>> tf.__version__
'2.6.0'
>>> exit()
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn# ls
ASAN.md     BUGS.md  EXTENDING.md  VERSION.md  prep-ivysyn-comparison.sh  pytorch-conda-requirements.txt  results                   src                          tf     venv
AUTHORS.md  COPYING  README.md     do-run.sh   prep-ivysyn.sh             pytorch-pip-requirements.txt    show-fuzzing-progress.py  tensorflow-requirements.txt  torch
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn# cd tf/
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
tf.audio.decode_wav-reported                                        tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceGather-known
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.StatefulPartitionedCall-reported
tf.linalg.solve-reported                                            tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorListConcat-reported
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.TensorScatterSub-confirmed
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.raw_ops.UnicodeEncode-confirmed
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_add-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyMomentum-reported               tf.tensor_scatter_nd_update-reported
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.audio.decode_wav-reported/bug.py 
import tensorflow as tf
import numpy as np
import io

# Generate input data
sample_rate = 44100
num_channels = 2
duration_seconds = 5
num_samples = sample_rate * duration_seconds
input_data = np.random.randint(-32768, 32767, size=(num_samples, num_channels)).astype(np.int16)

# Convert NumPy array to WAV file contents
wav_io = io.BytesIO()
wav_write = tf.audio.encode_wav(input_data, sample_rate)
wav_io.write(wav_write.numpy())
wav_contents = wav_io.getvalue()

# Decode the WAV file contents with desired_channels satisfying the requirement
desired_channels = -num_channels - 1  # int_1 is less than 0 and also less than negative of the number of dimensions
decoded_audio = tf.audio.decode_wav(wav_contents, desired_channels=desired_channels)

# Print the decoded audio tensor
print(decoded_audio)

# https://github.com/tensorflow/tensorflow/issues/63687
# int_1 is less than 0 and also less than negative of the number of dimensions(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.audio.decode_wav-reported/bug.py 
2024-03-16 04:29:30.909959: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:29:30.909989: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:29:31.604224: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:29:31.604275: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:29:31.604285: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:29:31.604378: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:29:31.604424: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:29:31.604644: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-16 04:29:31.627249: F tensorflow/core/framework/tensor_shape.cc:187] Non-OK-status: InitDims(dim_sizes) status: Internal: Expected a non-negative size, got -3
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.experimental.numpy.cos-confirmed/bug.py 
import tensorflow as tf

# Generate input data
input_data = tf.constant([0.0, 1.0, 2.0, 3.0, 4.0])

# Quantize the input data
quantized_input_data = tf.quantization.fake_quant_with_min_max_args(input_data, min=-6.0, max=6.0)

# Invoke tf.experimental.numpy.cos to process quantized input data
result = tf.experimental.numpy.cos(quantized_input_data)

print(result)

# https://github.com/tensorflow/tensorflow/issues/63271
# A tensor that is already quantized(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.experimental.numpy.cos-confirmed/bug.py 
2024-03-16 04:29:47.778265: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:29:47.778292: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:29:48.476263: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:29:48.476305: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:29:48.476313: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:29:48.476408: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:29:48.476454: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:29:48.476645: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
tf.Tensor([ 1.          0.5501642  -0.43742502 -0.99158424 -0.65364325], shape=(5,), dtype=float32)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
tf.audio.decode_wav-reported                                        tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceGather-known
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.StatefulPartitionedCall-reported
tf.linalg.solve-reported                                            tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorListConcat-reported
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.TensorScatterSub-confirmed
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.raw_ops.UnicodeEncode-confirmed
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_add-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyMomentum-reported               tf.tensor_scatter_nd_update-reported
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.image.draw_bounding_boxes-reported/bug.py 
import tensorflow as tf
import numpy as np

# Generate input data
batch_size = 1
image_height = 100
image_width = 100
num_channels = 3
num_boxes = 2

images = np.random.rand(batch_size, image_height, image_width, num_channels)
boxes = np.random.rand(batch_size, num_boxes, 4)
colors = np.array([[1.0, 0.0, 0.0], [0.0, 1.0, 0.0]])  # Define colors for the bounding boxes

# Invoke tf.image.draw_bounding_boxes
input_images = tf.convert_to_tensor(images, dtype=tf.float32)
input_boxes = tf.convert_to_tensor(boxes, dtype=tf.float32)
input_colors = tf.convert_to_tensor(colors, dtype=tf.float32)

# Reshape the input_images to satisfy the requirement
reshaped_images = tf.reshape(input_images, [image_height, image_width, num_channels])

output_images = tf.image.draw_bounding_boxes(reshaped_images, input_boxes, colors)

# Print the output images
print(output_images)

# https://github.com/tensorflow/tensorflow/issues/63688(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.image.draw_bounding_boxes-reported/bug.py 
2024-03-16 04:30:02.008190: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:30:02.008219: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:30:02.705315: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:30:02.705355: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:30:02.705363: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:30:02.705462: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:30:02.705504: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:30:02.705697: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Traceback (most recent call last):
  File "/home/ivysyn/tf/tf.image.draw_bounding_boxes-reported/bug.py", line 23, in <module>
    output_images = tf.image.draw_bounding_boxes(reshaped_images, input_boxes, colors)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/util/dispatch.py", line 206, in wrapper
    return target(*args, **kwargs)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/image_ops_impl.py", line 5654, in draw_bounding_boxes_v2
    return gen_image_ops.draw_bounding_boxes_v2(images, boxes, colors, name)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/gen_image_ops.py", line 1448, in draw_bounding_boxes_v2
    _ops.raise_from_not_ok_status(e, name)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 6941, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: The rank of the images should be 4 [Op:DrawBoundingBoxesV2]
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
tf.audio.decode_wav-reported                                        tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceGather-known
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.StatefulPartitionedCall-reported
tf.linalg.solve-reported                                            tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorListConcat-reported
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.TensorScatterSub-confirmed
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.raw_ops.UnicodeEncode-confirmed
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_add-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyMomentum-reported               tf.tensor_scatter_nd_update-reported
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.io.TFRecordOptions-confirmed/bug.py 
import tensorflow as tf

# Generate input data
input_data = b'input_data_example'

# Create TFRecordOptions with custom compression_method
options = tf.io.TFRecordOptions(compression_type='ZLIB', compression_method=3)

# Process input data using TFRecordOptions
with tf.io.TFRecordWriter('output.tfrecord', options=options) as writer:
    writer.write(input_data)

# https://github.com/tensorflow/tensorflow/issues/63337
# int (tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.io.TFRecordOptions-confirmed/bug.py 
2024-03-16 04:30:29.591599: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:30:29.591627: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:30:30.283814: F tensorflow/core/lib/io/record_writer.cc:74] Failed to initialize Zlib inputbuffer. Error: Invalid argument: deflateInit failed with status-2
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.linalg.solve-reported/bug.py 
import tensorflow as tf
import numpy as np

# Generate random input data
matrix = np.random.rand(2, 3)
rhs = np.random.rand(3)

# Invoke tf.linalg.solve
result = tf.linalg.solve(matrix, rhs)

# Print the result
print(result)

# https://github.com/tensorflow/tensorflow/issues/63339
# A 1-dimensional tensor (vector)(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.linalg.solve-reported/bug.py 
2024-03-16 04:30:55.041513: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:30:55.041540: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:30:55.737596: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:30:55.737637: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:30:55.737646: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:30:55.737740: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:30:55.737781: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:30:55.737967: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Traceback (most recent call last):
  File "/home/ivysyn/tf/tf.linalg.solve-reported/bug.py", line 9, in <module>
    result = tf.linalg.solve(matrix, rhs)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/gen_linalg_ops.py", line 1639, in matrix_solve
    return matrix_solve_eager_fallback(
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/gen_linalg_ops.py", line 1687, in matrix_solve_eager_fallback
    _result = _execute.execute(b"MatrixSolve", 1, inputs=_inputs_flat,
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/eager/execute.py", line 59, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.InvalidArgumentError: All input tensors must have the same rank. [Op:MatrixSolve]
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed/bug.py 
import tensorflow as tf

# Generate input data
input_data = tf.constant(3.0)

# Define min and max values per channel
min_per_channel = tf.constant(2.0)
max_per_channel = tf.constant(4.0)

# Invoke fake_quant_with_min_max_vars_per_channel
quantized_data = tf.quantization.fake_quant_with_min_max_vars_per_channel(input_data, min_per_channel, max_per_channel)

# Print the quantized data
print(quantized_data)

# https://github.com/tensorflow/tensorflow/issues/63351
# A zero-dimensional tensor (scalar)(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed/bug.py 
2024-03-16 04:31:10.062576: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:31:10.062612: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:31:10.759482: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:31:10.759522: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:31:10.759531: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:31:10.759623: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:31:10.759665: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:31:10.759857: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Traceback (most recent call last):
  File "/home/ivysyn/tf/tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed/bug.py", line 11, in <module>
    quantized_data = tf.quantization.fake_quant_with_min_max_vars_per_channel(input_data, min_per_channel, max_per_channel)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/gen_array_ops.py", line 3115, in fake_quant_with_min_max_vars_per_channel
    _ops.raise_from_not_ok_status(e, name)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 6941, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: min has incorrect size, expected 0 was 27096 [Op:FakeQuantWithMinMaxVarsPerChannel]
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.AudioSpectrogram-confirmed/bug.py 
import tensorflow as tf

# Generate input data
input_data = tf.random.normal([1, 44100], dtype=tf.float32)

# Invoke tf.raw_ops.AudioSpectrogram with a negative window_size
spectrogram = tf.raw_ops.AudioSpectrogram(input=input_data, window_size=-1024, stride=64, magnitude_squared=False)

# Print the spectrogram
print(spectrogram)

# https://github.com/tensorflow/tensorflow/issues/63352
# int_1 is a negative integer(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.AudioSpectrogram-confirmed/bug.py 
2024-03-16 04:31:25.554532: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:31:25.554560: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:31:26.249511: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:31:26.249551: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:31:26.249559: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:31:26.249654: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:31:26.249697: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:31:26.249888: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Segmentation fault (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.AvgPoolGrad-confirmed/bug.py 
import tensorflow as tf

# Generate input data
input_data = tf.random.normal([1, 28, 28, 3])
grad = tf.random.normal([1, 14, 14, 6])  # Change the number of channels in grad tensor

# Perform average pooling
result = tf.nn.avg_pool2d(input_data, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='VALID', data_format='NHWC')

# Compute gradient
grad_result = tf.raw_ops.AvgPoolGrad(orig_input_shape=tf.shape(input_data), grad=grad, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='VALID', data_format='NHWC')

print(grad_result)

# https://github.com/tensorflow/tensorflow/issues/63353
# change value(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.AvgPoolGrad-confirmed/bug.py 
2024-03-16 04:31:41.428067: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:31:41.428095: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:31:42.123886: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:31:42.123928: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:31:42.123936: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:31:42.124034: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:31:42.124075: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:31:42.124266: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
tf.Tensor(
[[[[-0.03918742  0.4218067  -0.03788855]
   [-0.5579095  -0.02110772 -0.42071307]
   [-0.03918742  0.4218067  -0.03788855]
   ...
   [-0.17213851  0.09030657  0.12213329]
   [ 0.3524588  -0.01084446 -0.1421077 ]
   [-0.17213851  0.09030657  0.12213329]]

  [[ 0.01803914 -0.20197551 -0.23984294]
   [-0.12727955  0.19848748  0.08953874]
   [ 0.01803914 -0.20197551 -0.23984294]
   ...
   [ 0.03274738  0.0139521   0.24972066]
   [-0.04758611 -0.13721116 -0.14140286]
   [ 0.03274738  0.0139521   0.24972066]]

  [[-0.03918742  0.4218067  -0.03788855]
   [-0.5579095  -0.02110772 -0.42071307]
   [-0.03918742  0.4218067  -0.03788855]
   ...
   [-0.17213851  0.09030657  0.12213329]
   [ 0.3524588  -0.01084446 -0.1421077 ]
   [-0.17213851  0.09030657  0.12213329]]

  ...

  [[ 0.05686652  0.12444445  0.4328659 ]
   [ 0.30154216  0.20104063 -0.02151499]
   [ 0.05686652  0.12444445  0.4328659 ]
   ...
   [-0.0204222   0.1683027   0.28603598]
   [-0.08124992  0.00556594  0.47688538]
   [-0.0204222   0.1683027   0.28603598]]

  [[ 0.23122242  0.44934702  0.22925375]
   [-0.03096152 -0.20691301  0.11211004]
   [ 0.23122242  0.44934702  0.22925375]
   ...
   [ 0.68494254 -0.12935176  0.24804828]
   [-0.21339597  0.18001083  0.27241448]
   [ 0.68494254 -0.12935176  0.24804828]]

  [[ 0.05686652  0.12444445  0.4328659 ]
   [ 0.30154216  0.20104063 -0.02151499]
   [ 0.05686652  0.12444445  0.4328659 ]
   ...
   [-0.0204222   0.1683027   0.28603598]
   [-0.08124992  0.00556594  0.47688538]
   [-0.0204222   0.1683027   0.28603598]]]], shape=(1, 28, 28, 3), dtype=float32)
free(): invalid next size (normal)
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.DatasetToTFRecord-reported/bug.py 
import tensorflow as tf

# Generate input data
input_data = [[1, 2], [3, 4], [5, 6]]

# Convert input data to strings
input_data_strings = [[str(d) for d in inner_list] for inner_list in input_data]

# Create a dataset from the input data strings
dataset = tf.data.Dataset.from_tensor_slices(input_data_strings)

# Define the filename for the TFRecord file
filename = "output.tfrecord"

# Invoke tf.raw_ops.DatasetToTFRecord to process input data
tf.raw_ops.DatasetToTFRecord(input_dataset=dataset._variant_tensor, filename=filename, compression_type="")

# https://github.com/tensorflow/tensorflow/issues/63689
# A 2D tensor on a CPU backend(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.DatasetToTFRecord-reported/bug.py 
2024-03-16 04:32:01.600767: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:32:01.600797: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:32:02.291302: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:32:02.291343: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:32:02.291353: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:32:02.291447: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:32:02.291490: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:32:02.291693: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-16 04:32:02.315377: F tensorflow/core/framework/tensor.cc:681] Check failed: 1 == NumElements() (1 vs. 2)Must have a one element tensor
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.Dilation2DBackpropFilter-known/bug.py 
import tensorflow as tf

# Generate input data
input_data = tf.random.normal([1, 28, 28, 3])
filter_data = tf.random.normal([3, 3, 3])
out_backprop_data = tf.random.normal([1, 28, 28, 3, 3])  # Adding an extra dimension

# Invoke tf.raw_ops.Dilation2DBackpropFilter
output = tf.raw_ops.Dilation2DBackpropFilter(input=input_data, filter=filter_data, out_backprop=out_backprop_data, strides=[1, 1, 1, 1], rates=[1, 1, 1, 1], padding="SAME")

print(output)

# known
# shape(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.Dilation2DBackpropFilter-known/bug.py 
2024-03-16 04:32:23.524760: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:32:23.524788: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:32:24.219416: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:32:24.219457: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:32:24.219467: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:32:24.219560: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:32:24.219600: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:32:24.219786: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-16 04:32:24.234543: F tensorflow/core/framework/tensor_shape.cc:46] Check failed: NDIMS == dims() (4 vs. 5)Asking for tensor of 4 dimensions from a tensor of 5 dimensions
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.DrawBoundingBoxes
tf.raw_ops.DrawBoundingBoxes-known/   tf.raw_ops.DrawBoundingBoxesV2-known/ 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.DrawBoundingBoxes-known/bug.py 
import tensorflow as tf
import numpy as np

# Generate input data
batch_size = 1
image_height = 100
image_width = 100
num_channels = 3
num_boxes = 2

images = np.random.rand(image_height, image_width, num_channels).astype(np.float32)  # Remove the batch dimension
boxes = np.random.rand(batch_size, num_boxes, 4).astype(np.float32)

# Invoke tf.raw_ops.DrawBoundingBoxes with a zero-dimensional tensor for images
drawn_images = tf.raw_ops.DrawBoundingBoxes(images=tf.convert_to_tensor(images),
                                           boxes=tf.convert_to_tensor(boxes))

# Print the result
print(drawn_images)

# https://github.com/tensorflow/tensorflow/issues/63354
# A tensor with 3 dimensions(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.DrawBoundingBoxes-known/bug.py 
2024-03-16 04:32:44.612213: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:32:44.612243: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:32:45.309948: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:32:45.309988: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:32:45.309996: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:32:45.310089: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:32:45.310131: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:32:45.310327: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Traceback (most recent call last):
  File "/home/ivysyn/tf/tf.raw_ops.DrawBoundingBoxes-known/bug.py", line 15, in <module>
    drawn_images = tf.raw_ops.DrawBoundingBoxes(images=tf.convert_to_tensor(images),
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/util/tf_export.py", line 404, in wrapper
    return f(**kwargs)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/gen_image_ops.py", line 1375, in draw_bounding_boxes
    _ops.raise_from_not_ok_status(e, name)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 6941, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: The rank of the images should be 4 [Op:DrawBoundingBoxes]
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.DrawBoundingBoxesV2-known/bug.py 
import tensorflow as tf
import numpy as np

# Generate input data
image_height = 100
image_width = 100
num_channels = 3
num_boxes = 2

images = tf.random.uniform((image_height, image_width, num_channels))  # Change the shape to satisfy the requirement of a zero-dimensional tensor
boxes = tf.random.uniform((1, num_boxes, 4))
colors = tf.constant([[1.0, 0.0, 0.0], [0.0, 1.0, 0.0]])  # Define colors for each bounding box

# Invoke tf.raw_ops.DrawBoundingBoxesV2
output_images = tf.raw_ops.DrawBoundingBoxesV2(images=images, boxes=boxes, colors=colors)

# Print the output images
print(output_images)

# https://github.com/tensorflow/tensorflow/issues/63354(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.DrawBoundingBoxesV2-known/bug.py 
2024-03-16 04:32:57.983363: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:32:57.983389: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:32:58.679623: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:32:58.679664: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:32:58.679677: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:32:58.679771: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:32:58.679813: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:32:58.680006: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Traceback (most recent call last):
  File "/home/ivysyn/tf/tf.raw_ops.DrawBoundingBoxesV2-known/bug.py", line 15, in <module>
    output_images = tf.raw_ops.DrawBoundingBoxesV2(images=images, boxes=boxes, colors=colors)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/util/tf_export.py", line 404, in wrapper
    return f(**kwargs)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/gen_image_ops.py", line 1448, in draw_bounding_boxes_v2
    _ops.raise_from_not_ok_status(e, name)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 6941, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: The rank of the images should be 4 [Op:DrawBoundingBoxesV2]
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.ExperimentalDatasetToTFRecord-reported/bug.py 
import tensorflow as tf

# Generate input data
input_data = tf.data.Dataset.range(10).map(lambda x: tf.strings.as_string(x))
input_data = input_data.batch(2)  # Batch the dataset to have 3 dimensions with each dimension size greater than 1

# Invoke tf.raw_ops.ExperimentalDatasetToTFRecord to process input data
filename = tf.constant("output.tfrecord")
compression_type = tf.constant("GZIP")
tf.raw_ops.ExperimentalDatasetToTFRecord(input_dataset=input_data._variant_tensor, filename=filename, compression_type=compression_type)

# https://github.com/tensorflow/tensorflow/issues/63691
# A tensor with 3 dimensions, each dimension of size greater than 1(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.ExperimentalDatasetToTFRecord-reported/bug.py 
2024-03-16 04:33:14.350773: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:33:14.350802: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:33:15.043315: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:33:15.043356: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:33:15.043364: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:33:15.043460: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:33:15.043502: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:33:15.043700: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-16 04:33:15.090620: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:185] None of the MLIR Optimization Passes are enabled (registered 2)
2024-03-16 04:33:15.097772: F tensorflow/core/framework/tensor.cc:681] Check failed: 1 == NumElements() (1 vs. 2)Must have a one element tensor
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed/bug.py 
import tensorflow as tf

# Generate input data
input_data = tf.constant([[1.5, 2.5, 3.5], [4.5, 5.5, 6.5]])

# Define min and max values per channel
min_per_channel = tf.constant([1.0, 2.0, 3.0])
max_per_channel = tf.constant([2.0, 3.0, 4.0])

# Invoke tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel with inputs as 0-dimensional tensor and max as a 1x3 tensor
quantized_output = tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel(inputs=tf.constant(0.0), min=min_per_channel, max=max_per_channel, num_bits=8, narrow_range=False)

# Print the quantized output
print(quantized_output)

# https://github.com/tensorflow/tensorflow/issues/63355
# 0-dimensional tensor(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed/bug.py 
2024-03-16 04:33:27.997918: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:33:27.997947: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:33:28.683584: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:33:28.683624: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:33:28.683632: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:33:28.683726: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:33:28.683768: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:33:28.683960: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Traceback (most recent call last):
  File "/home/ivysyn/tf/tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed/bug.py", line 11, in <module>
    quantized_output = tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel(inputs=tf.constant(0.0), min=min_per_channel, max=max_per_channel, num_bits=8, narrow_range=False)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/util/tf_export.py", line 404, in wrapper
    return f(**kwargs)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/gen_array_ops.py", line 3115, in fake_quant_with_min_max_vars_per_channel
    _ops.raise_from_not_ok_status(e, name)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 6941, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: min has incorrect size, expected 0 was 3 [Op:FakeQuantWithMinMaxVarsPerChannel]
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.FusedPadConv2D-confirmed/bug.py 
import tensorflow as tf

# Generate input data
input_data = tf.random.normal([3, 10, 10])

# Define paddings
paddings = tf.constant([[0, 0], [1, 1], [1, 1]])

# Define filter
filter = tf.random.normal([3, 3, 3, 16])

# Define mode
mode = "REFLECT"  # Change mode to "REFLECT" or "SYMMETRIC"

# Define strides
strides = [1, 1, 1, 1]

# Define padding
padding = "VALID"

# Invoke tf.raw_ops.FusedPadConv2D
output = tf.raw_ops.FusedPadConv2D(input=input_data, paddings=paddings, filter=filter, mode=mode, strides=strides, padding=padding)

print(output)

# https://github.com/tensorflow/tensorflow/issues/63376
# A tensor with 3 dimensions, each dimension of size greater than 1(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.FusedPadConv2D-confirmed/bug.py 
2024-03-16 04:33:52.156803: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:33:52.156832: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:33:52.859957: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:33:52.859997: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:33:52.860005: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:33:52.860100: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:33:52.860143: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:33:52.860337: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Traceback (most recent call last):
  File "/home/ivysyn/tf/tf.raw_ops.FusedPadConv2D-confirmed/bug.py", line 22, in <module>
    output = tf.raw_ops.FusedPadConv2D(input=input_data, paddings=paddings, filter=filter, mode=mode, strides=strides, padding=padding)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/util/tf_export.py", line 404, in wrapper
    return f(**kwargs)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/gen_nn_ops.py", line 4388, in fused_pad_conv2d
    _ops.raise_from_not_ok_status(e, name)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 6941, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: The first dimension of paddings must be the rank of inputs: 4 [3,2] [3,10,10,0] [Op:FusedPadConv2D]
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.QuantizeAndDequantizeV3-reported/bug.py 
import tensorflow as tf

# Generate input data
input_data = tf.constant([1.5, 2.5, 3.5, 4.5])

# Define input_min with 3 dimensions, each dimension of size greater than 1
input_min = tf.constant([[[0.0, 0.0], [0.0, 0.0]], [[0.0, 0.0], [0.0, 0.0]], [[0.0, 0.0], [0.0, 0.0]]])

# Define input_max, num_bits, and other parameters
input_max = tf.constant(5.0)
num_bits = tf.constant(8)
signed_input = True
range_given = True
narrow_range = False
axis = -1

# Invoke tf.raw_ops.QuantizeAndDequantizeV3
output_data = tf.raw_ops.QuantizeAndDequantizeV3(input=input_data, input_min=input_min, input_max=input_max, num_bits=num_bits, signed_input=signed_input, range_given=range_given, narrow_range=narrow_range, axis=axis)

# Print the output data
print(output_data)

# https://github.com/tensorflow/tensorflow/issues/63693
# A tensor with 3 dimensions, each dimension of size greater than 1(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.QuantizeAndDequantizeV3-reported/bug.py 
2024-03-16 04:34:02.583436: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:34:02.583463: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:34:03.233337: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:34:03.233378: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:34:03.233386: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:34:03.233482: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:34:03.233527: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:34:03.233724: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-16 04:34:03.248029: F tensorflow/core/framework/tensor.cc:681] Check failed: 1 == NumElements() (1 vs. 12)Must have a one element tensor
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.ResourceApplyGradientDescent-reported/bug.py 
import tensorflow as tf

# Generate input data
var = tf.Variable(3.0 + 1j, dtype=tf.complex64)  # Change the dtype to complex
alpha = tf.constant(0.1, dtype=tf.float32)
delta = tf.constant(0.5, dtype=tf.float32)

# Invoke tf.raw_ops.ResourceApplyGradientDescent
result = tf.raw_ops.ResourceApplyGradientDescent(var=var.handle, alpha=alpha, delta=delta)

# Print the result
print(result)

# https://github.com/tensorflow/tensorflow/issues/63695
# A complex tensor(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.ResourceApplyGradientDescent-reported/bug.py 
2024-03-16 04:34:14.724390: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:34:14.724416: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:34:15.420365: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:34:15.420403: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:34:15.420411: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:34:15.420507: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:34:15.420548: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:34:15.420694: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-16 04:34:15.435392: F tensorflow/core/framework/tensor.cc:667] Check failed: dtype() == expected_dtype (8 vs. 1) float expected, got complex64
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.ResourceApplyMomentum-reported/bug.py 
import tensorflow as tf

# Generate input data
var = tf.Variable([1.0, 2.0, 3.0])
accum = tf.Variable([0.1, 0.2, 0.3], dtype=tf.complex64)  # Change the dtype to complex
lr = 0.01
grad = tf.constant([0.1, 0.2, 0.3])
momentum = 0.9

# Invoke tf.raw_ops.ResourceApplyMomentum with correct input types
output = tf.raw_ops.ResourceApplyMomentum(var=var.handle, accum=accum.handle, lr=lr, grad=grad, momentum=momentum)

# Print the output
print(output)

# https://github.com/tensorflow/tensorflow/issues/63696
# A complex tensor(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.ResourceApplyMomentum-reported/bug.py 
2024-03-16 04:34:30.223303: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:34:30.223329: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:34:30.923347: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:34:30.923386: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:34:30.923394: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:34:30.923486: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:34:30.923530: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:34:30.923685: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-16 04:34:30.938765: F tensorflow/core/framework/tensor.cc:673] Check failed: dtype() == expected_dtype (8 vs. 1) float expected, got complex64
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.ResourceApplyProximalAdagrad-confirmed/bug.py 
import tensorflow as tf

# Generate input data
var = tf.Variable([1.0, 2.0, 3.0])
accum = tf.Variable([0.1, 0.2, 0.3])
lr = 0.01
l1 = 0.1
l2 = 0.01
grad = tf.constant([0.1, 0.2, 0.3], dtype=tf.float64)  # Change the dtype of grad to satisfy the requirement

# Convert var and accum to resource tensors
var_resource = tf.raw_ops.VarHandleOp(dtype=var.dtype, shape=var.shape)
accum_resource = tf.raw_ops.VarHandleOp(dtype=accum.dtype, shape=accum.shape)

# Assign var and accum to the resource tensors
assign_var = tf.raw_ops.AssignVariableOp(resource=var_resource, value=var)
assign_accum = tf.raw_ops.AssignVariableOp(resource=accum_resource, value=accum)

# Invoke tf.raw_ops.ResourceApplyProximalAdagrad with resource tensors
result = tf.raw_ops.ResourceApplyProximalAdagrad(var=var_resource, accum=accum_resource, lr=lr, l1=l1, l2=l2, grad=grad)

# Print the result
print(result)

# https://github.com/tensorflow/tensorflow/issues/63697
# A tensor that is in floating-point format(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.ResourceApplyProximalAdagrad-confirmed/bug.py 
2024-03-16 04:35:01.768500: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:35:01.768531: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:35:02.465523: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:35:02.465560: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:35:02.465570: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:35:02.465668: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:35:02.465709: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:35:02.465857: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-16 04:35:02.481384: F tensorflow/core/framework/tensor.cc:667] Check failed: dtype() == expected_dtype (1 vs. 2) double expected, got float
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.ResourceGather-known/bug.py 
import tensorflow as tf

# Generate input data
input_data = tf.constant([[1, 2], [3, 4], [5, 6]])
indices = tf.constant([0, 1])

# Create a resource variable on a different device and with a different dtype
with tf.device('/device:GPU:0'):
    resource_var = tf.Variable([7, 8, 9], dtype=tf.float32)

# Use tf.raw_ops.ResourceGather to gather values
output = tf.raw_ops.ResourceGather(resource=resource_var.handle, indices=indices, dtype=tf.int32)

print(output)

# known bug
# tensor_1 is on a different device than the input device and has a different dtype(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.ResourceGather-known/bug.py 
2024-03-16 04:35:15.794740: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:35:15.794768: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:35:16.498266: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:35:16.498306: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:35:16.498313: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:35:16.498407: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:35:16.498449: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:35:16.498642: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-16 04:35:16.513919: F tensorflow/core/framework/tensor.cc:667] Check failed: dtype() == expected_dtype (1 vs. 3) int32 expected, got float
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.ResourceSparseApplyKerasMomentum-reported/bug.py 
import tensorflow as tf

# Generate input data
var = tf.Variable([[1.0, 2.0], [3.0, 4.0]], dtype=tf.complex64)  # Change var to a complex tensor
accum = tf.Variable([[0.1, 0.2], [0.3, 0.4]])
lr = 0.01
grad = tf.constant([[0.1, 0.2], [0.3, 0.4]])
indices = tf.constant([0, 1])  # Make indices one-dimensional
momentum = 0.9

# Invoke tf.raw_ops.ResourceSparseApplyKerasMomentum
result = tf.raw_ops.ResourceSparseApplyKerasMomentum(var=var.handle, accum=accum.handle, lr=lr, grad=grad, indices=indices, momentum=momentum)

# Print the result
print(result)

# https://github.com/tensorflow/tensorflow/issues/63720
# A complex tensor(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.ResourceSparseApplyKerasMomentum-reported/bug.py 
2024-03-16 04:35:29.379052: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:35:29.379082: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:35:30.070687: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:35:30.070726: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:35:30.070736: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:35:30.070829: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:35:30.070873: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:35:30.071036: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-16 04:35:30.086309: F tensorflow/core/framework/tensor.cc:673] Check failed: dtype() == expected_dtype (8 vs. 1) float expected, got complex64
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cd tf.raw_ops.StatefulPartitionedCall-reported/bug.py 
bash: cd: tf.raw_ops.StatefulPartitionedCall-reported/bug.py: Not a directory
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.StatefulPartitionedCall-reported/bug.py 
import tensorflow as tf

# Generate input data
input_data = tf.constant([[1, 2, 3], [4, 5, 6]])

# Define the function to be called
@tf.function
def my_function(inputs):
    return tf.reduce_sum(inputs, axis=1)

# Create a callable from the function
callable_function = tf.function(my_function).get_concrete_function(tf.TensorSpec(shape=None, dtype=tf.int32))

# Invoke tf.raw_ops.StatefulPartitionedCall to process input data
output = tf.raw_ops.StatefulPartitionedCall(args=[input_data], Tout=None, f=callable_function)

# Print the output
print(output)

# https://github.com/tensorflow/tensorflow/issues/63721
# tensor_1 is an undefined or None value(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.StatefulPartitionedCall-reported/bug.py 
2024-03-16 04:35:46.811945: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:35:46.811974: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:35:47.498175: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:35:47.498216: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:35:47.498224: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:35:47.498317: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:35:47.498360: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:35:47.498549: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-16 04:35:47.544727: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:185] None of the MLIR Optimization Passes are enabled (registered 2)
2024-03-16 04:35:47.550819: F tensorflow/core/framework/op_kernel.cc:898] Check failed: index < outputs_.size() (0 vs. 0)
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.TensorListConcat-reported/bug.py 
import tensorflow as tf

# Generate input data
input_data_1 = tf.constant([[1, 2, 3], [4, 5, 6]])
input_data_2 = tf.constant([[7, 8, 9], [10, 11, 12]])

# Create a tensor list
input_handle = tf.raw_ops.TensorListReserve(element_dtype=tf.int32, element_shape=[2, 3], num_elements=2)
input_handle = tf.raw_ops.TensorListSetItem(input_handle=input_handle, index=0, item=input_data_1)
input_handle = tf.raw_ops.TensorListSetItem(input_handle=input_handle, index=1, item=input_data_2)

# Invoke TensorListConcat with explicit shape
concatenated_data, lengths = tf.raw_ops.TensorListConcat(input_handle=input_handle, element_dtype=tf.int32, element_shape=[6])

# Print the concatenated data
print(concatenated_data)

# https://github.com/tensorflow/tensorflow/issues/63722
# A 1-dimensional tensor (vector)(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.TensorListConcat-reported/bug.py 
2024-03-16 04:35:58.731482: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:35:58.731510: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:35:59.424315: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:35:59.424356: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:35:59.424364: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:35:59.424459: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:35:59.424503: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:35:59.424697: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
tf.Tensor([1 2 3 4], shape=(4,), dtype=int32)
free(): invalid next size (fast)
Aborted (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.TensorScatterSub-confirmed/bug.py 
import tensorflow as tf

# Generate input data
tensor = tf.constant([1, 2, 3, 4, 5])
indices = tf.constant([[[1], [3]], [[0], [2]]])  # Nested structure for indices
updates = tf.constant([10, 20])

# Invoke tf.raw_ops.TensorScatterSub
result = tf.raw_ops.TensorScatterSub(tensor=tensor, indices=indices, updates=updates)

# Print the result
print(result)

# https://github.com/tensorflow/tensorflow/issues/63378
# A tensor that contains other tensors, creating a nested structure.(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.TensorScatterSub-confirmed/bug.py 
2024-03-16 04:36:11.981583: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:36:11.981614: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:36:12.676200: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:36:12.676241: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:36:12.676249: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:36:12.676343: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:36:12.676384: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:36:12.676576: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Traceback (most recent call last):
  File "/home/ivysyn/tf/tf.raw_ops.TensorScatterSub-confirmed/bug.py", line 9, in <module>
    result = tf.raw_ops.TensorScatterSub(tensor=tensor, indices=indices, updates=updates)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/util/tf_export.py", line 404, in wrapper
    return f(**kwargs)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/gen_array_ops.py", line 11185, in tensor_scatter_sub
    _ops.raise_from_not_ok_status(e, name)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 6941, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: Outer dimensions of indices and update must match. Indices shape: [2,2,1], updates shape:[2] [Op:TensorScatterSub]
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.raw_ops.UnicodeEncode-confirmed/bug.py 
import tensorflow as tf

# Generate input data
input_values = tf.constant([72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100])  # Unicode codepoints for "Hello World"
input_splits = tf.constant([[0, 5, 11]])  # Split indices for the input_values with two dimensions
output_encoding = "UTF-8"

# Invoke tf.raw_ops.unicode_encode
output = tf.raw_ops.UnicodeEncode(input_values=input_values, input_splits=input_splits, output_encoding=output_encoding)

# Print the output
print(output)

# https://github.com/tensorflow/tensorflow/issues/63379
# A 2D tensor on a CPU backend(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.raw_ops.UnicodeEncode-confirmed/bug.py 
2024-03-16 04:36:24.274728: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:36:24.274759: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:36:24.966523: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:36:24.966565: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:36:24.966573: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:36:24.966667: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:36:24.966710: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:36:24.966905: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Segmentation fault (core dumped)
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.tensor_scatter_nd_add-reported/bug.py 
import tensorflow as tf

# Generate input data
input_tensor = tf.zeros([15, 15, 15])
indices = tf.constant([[[0, 0, 0], [1, 1, 1]], [[2, 2, 2], [3, 3, 3]], [[4, 4, 4], [5, 5, 5]], [[6, 6, 6], [7, 7, 7]], [[8, 8, 8], [9, 9, 9]], [[10, 10, 10], [11, 11, 11]], [[12, 12, 12], [13, 13, 13]], [[14, 14, 14], [0, 0, 0]], [[1, 1, 1], [2, 2, 2]], [[3, 3, 3], [4, 4, 4]], [[5, 5, 5], [6, 6, 6]], [[7, 7, 7], [8, 8, 8]], [[9, 9, 9], [10, 10, 10]], [[11, 11, 11], [12, 12, 12]], [[13, 13, 13], [14, 14, 14]]])
updates = tf.constant([1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0])  # Cast updates to float

# Invoke tf.tensor_scatter_nd_add
result = tf.tensor_scatter_nd_add(input_tensor, indices, updates)

# Print the result
print(result)

# https://github.com/tensorflow/tensorflow/issues/63380
# A tensor with shape (15, 15, 15) which has both a trailing dimension and total number of elements not divisible by 16(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.tensor_scatter_nd_add-reported/bug.py 
2024-03-16 04:36:40.891520: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:36:40.891548: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:36:41.588458: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:36:41.588500: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:36:41.588511: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:36:41.588607: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:36:41.588655: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:36:41.588848: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Traceback (most recent call last):
  File "/home/ivysyn/tf/tf.tensor_scatter_nd_add-reported/bug.py", line 9, in <module>
    result = tf.tensor_scatter_nd_add(input_tensor, indices, updates)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/gen_array_ops.py", line 10878, in tensor_scatter_add
    _ops.raise_from_not_ok_status(e, name)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 6941, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: Outer dimensions of indices and update must match. Indices shape: [15,2,3], updates shape:[15] [Op:TensorScatterAdd]
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# ls
output.tfrecord                                                     tf.raw_ops.Dilation2DBackpropFilter-known               tf.raw_ops.ResourceGather-known
tf.audio.decode_wav-reported                                        tf.raw_ops.DrawBoundingBoxes-known                      tf.raw_ops.ResourceSparseApplyKerasMomentum-reported
tf.experimental.numpy.cos-confirmed                                 tf.raw_ops.DrawBoundingBoxesV2-known                    tf.raw_ops.StatefulPartitionedCall-reported
tf.image.draw_bounding_boxes-reported                               tf.raw_ops.ExperimentalDatasetToTFRecord-reported       tf.raw_ops.TensorListConcat-reported
tf.io.TFRecordOptions-confirmed                                     tf.raw_ops.FakeQuantWithMinMaxVarsPerChannel-confirmed  tf.raw_ops.TensorScatterSub-confirmed
tf.linalg.solve-reported                                            tf.raw_ops.FusedPadConv2D-confirmed                     tf.raw_ops.UnicodeEncode-confirmed
tf.quantization.fake_quant_with_min_max_vars_per_channel-confirmed  tf.raw_ops.QuantizeAndDequantizeV3-reported             tf.tensor_scatter_nd_add-reported
tf.raw_ops.AudioSpectrogram-confirmed                               tf.raw_ops.ResourceApplyGradientDescent-reported        tf.tensor_scatter_nd_update-reported
tf.raw_ops.AvgPoolGrad-confirmed                                    tf.raw_ops.ResourceApplyMomentum-reported
tf.raw_ops.DatasetToTFRecord-reported                               tf.raw_ops.ResourceApplyProximalAdagrad-confirmed
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# cat tf.tensor_scatter_nd_update-reported/bug.py 
import tensorflow as tf

# Generate input data
input_tensor = tf.zeros([2, 2, 2])  # A tensor that contains other tensors, creating a nested structure
indices = tf.constant([[[0, 0, 0], [1, 1, 1]], [[1, 0, 1], [0, 1, 0]]])
updates = tf.constant([1, 2], dtype=tf.float32)  # Cast updates to float

# Invoke tf.tensor_scatter_nd_update
result = tf.tensor_scatter_nd_update(input_tensor, indices, updates)

# Print the result
print(result)

# https://github.com/tensorflow/tensorflow/issues/63375
# A tensor that contains other tensors, creating a nested structure.(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# python tf.tensor_scatter_nd_update-reported/bug.py 
2024-03-16 04:36:55.563637: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2024-03-16 04:36:55.563667: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2024-03-16 04:36:56.260805: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: UNKNOWN ERROR (34)
2024-03-16 04:36:56.260847: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: 5d1140ea8826
2024-03-16 04:36:56.260855: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: 5d1140ea8826
2024-03-16 04:36:56.260947: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:200] libcuda reported version is: Not found: was unable to find libcuda.so DSO loaded into this program
2024-03-16 04:36:56.260992: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:204] kernel reported version is: 535.161.7
2024-03-16 04:36:56.261185: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Traceback (most recent call last):
  File "/home/ivysyn/tf/tf.tensor_scatter_nd_update-reported/bug.py", line 9, in <module>
    result = tf.tensor_scatter_nd_update(input_tensor, indices, updates)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/util/dispatch.py", line 206, in wrapper
    return target(*args, **kwargs)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/array_ops.py", line 5754, in tensor_scatter_nd_update
    return gen_array_ops.tensor_scatter_update(
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/ops/gen_array_ops.py", line 11300, in tensor_scatter_update
    _ops.raise_from_not_ok_status(e, name)
  File "/home/ivysyn/venv/tensorflow-2.6-orig/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 6941, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: Outer dimensions of indices and update must match. Indices shape: [2,2,3], updates shape:[2] [Op:TensorScatterUpdate]
(tensorflow-2.6-orig) root@5d1140ea8826:/home/ivysyn/tf# 
